proposals/infra/INF-0004-register-types-and-diagnostics.md

```diff
@@ -129,97 +137,165 @@ final case will be entered. Types that are or contain a resource are known as
 "intangible". In this case, we are dealing with types that cannot be intangible.
 Types that can be immediately determined to not be intangible (that is, types that
 cannot be a resource type or cannot contain a resource type) are types like booleans,
-int, float, float4, etc. If the resource type is among any numerics or a type that
-cannot be an intangible type, there is only one exception where it can be treated
-as a resource. The only valid register type for such resource types is 'b', which
-adheres to legacy behavior. All other register types applied to such a type will 
-emit an error diagnostic that "the given type '%0' cannot be bound as a resource".
-If the register type is 'b', a warning will be emitted instead, that will be
-treated as an error by default: "register type 'b' used for resource type '%0', 
-which cannot be used as a resource". Below are some examples:
+int, float, float4, etc. If the variable type is among any numerics or a type that
+cannot be an intangible type, there is only one exception where the register keyword
+can legally be applied to the variable. The only valid register type for such a 
+numeric variable type is 'c'. The register keyword in this statement:
+`float f : register(c0)`
+isn't binding a resource, rather it is specifying a constant register binding offset
+within the $Globals cbuffer, which is legacy behavior from DX9. All other register types

```

Comment:
Another note here:
The context of the declaration where this is allowed is restricted to constants that would be placed into `$Globals` because they are outside of any `cbuffer {}` or `tbuffer {}` declaration.  In DXC and FXC, when the register binding is used in these other contexts it is silently ignored.  We should decide whether we want a DefaultError warning or hard error in that case.

Reply:
I think default error warning for now, and we can make it more strict and make it an error in 202x.